depthai_ros_driver/include/depthai_ros_driver/depthai_base.hpp

```diff
@@ -42,39 +104,27 @@ class DepthAIBase : public Node {
     ~DepthAIBase() = default;
 
 private:
-    enum Stream : std::size_t {
-        // all images at the beginning
-        LEFT,
-        RIGHT,
-        RECTIFIED_LEFT,
-        RECTIFIED_RIGHT,
-        DISPARITY,
-        DISPARITY_COLOR,
-        DEPTH,
-        PREVIEW_OUT,
-        // compressed image streams
-        JPEG_OUT,
-        VIDEO,
-        // non-image streams
-        META_D2H,
-        META_OUT,
-        OBJECT_TRACKER,
-        // utility enums
-        END,
-        IMAGE_END = META_D2H,
-        UNCOMPRESSED_IMG_END = JPEG_OUT
-    };
-    std::array<std::string, Stream::END> _stream_name{"left", "right", "rectified_left", "rectified_right", "disparity",
-            "disparity_color", "depth", "previewout", "jpegout", "video", "meta_d2h", "metaout", "object_tracker"};
-    std::array<std::string, Stream::END> _topic_name{"left", "right", "rectified_left", "rectified_right", "disparity",
-            "disparity_color", "depth", "previewout", "jpeg", "mjpeg", "meta_d2h", "object_info", "object_tracker"};
 
+    std::set<std::string> _enabled_streams;
     std::array<std::unique_ptr<ros::Publisher>, Stream::END> _stream_publishers;
     std::array<std::unique_ptr<ros::Publisher>, Stream::IMAGE_END> _camera_info_publishers;
 
     std::array<std::unique_ptr<camera_info_manager::CameraInfoManager>, Stream::IMAGE_END> _camera_info_manager;
+
     std::unique_ptr<camera_info_manager::CameraInfoManager> _defaultManager;
-    ros::ServiceServer _camera_info_default;
+    // ros::ServiceServer _camera_info_default;
+
+    std::unique_ptr<pluginlib::ClassLoader<rr::Pipeline> > _pipeline_loader;
```

Comment:
I don't think this has to be a class member

Reply:
this error is issued by making the pipeline loader to be local variable.
```
Warning: class_loader.ClassLoader: SEVERE WARNING!!! Attempting to unload library while objects created by this loader exist in the heap! You should delete your objects before attempting to unload the library or destroying the ClassLoader. The library will NOT be unloaded.
         at line 122 in /tmp/binarydeb/ros-melodic-class-loader-0.4.1/src/class_loader.cpp
```

Reply:
Ah, in that case, we can make it into a variable, instead of a pointer

Reply:
Still the same to make the pointer to the variable. 
It seems all the loaded plugins need to be unloaded for safely releasing the library.
http://docs.ros.org/en/jade/api/class_loader/html/class__loader_8cpp_source.html
http://docs.ros.org/en/jade/api/class_loader/html/class__loader_8h_source.html